### ISSUE TITLE
Typescript transpilation is "essentially broken" due to change in location of core library definitions

### ISSUE THREAD

**Issue by mattdiamond** [atom/atom#15909]
From [this comment](https://github.com/jakesankey/line-diff-details/issues/45#issuecomment-336309778):

> ...this has to do with the built-in TypeScript support that Atom has, which utilizes typescript-simple =1.0.0, which in turn depends on typescript ~1.4.1. The location of the core library definitions was moved from bin/ to lib/ somewhere between TypeScript 1.4 and 1.5, which means that typescript-simple is unable to find the core library defs (lib.d.ts) when installed with the most recent version of TypeScript matching ~1.4.1. In other words, Atom's built-in TypeScript support is essentially broken, and when it sees lib/line-diff-details.ts, it tries to transpile it, but the definitions it needs aren't where it expects them, and boom, error.

This issue is breaking plugins that depend on TypeScript transpilation, such as [line-diff-details](https://atom.io/packages/line-diff-details). According to the same user, the fix [should be simple](https://github.com/jakesankey/line-diff-details/issues/45#issuecomment-337011435):

> I think it should just require bumping Atom's dependency on typescript-simple from 1.0.0 (which is now broken) to any version that uses TypeScript 1.5 or later. It should probably not be 2.x, though, since that has the potential to leave some packages already using TS just as broken as they were before.

The exception:

```
At ENOENT, node_modules/typescript/bin/lib.d.ts not found in /Applications/Atom.app/Contents/Resources/app.asar

Error: ENOENT, node_modules/typescript/bin/lib.d.ts not found in /Applications/Atom.app/Contents/Resources/app.asar
    at notFoundError (ELECTRON_ASAR.js:115:19)
    at Object.fs.readFileSync (ELECTRON_ASAR.js:514:9)
    at TypeScriptSimple.createService (/Applications/Atom.app/Contents/Resources/app/node_modules/typescript-simple/index.js:71:73)
    at TypeScriptSimple.compile (/Applications/Atom.app/Contents/Resources/app/node_modules/typescript-simple/index.js:60:43)
    at Object.exports.compile (/Applications/Atom.app/Contents/Resources/app/src/typescript.js:54:53)
    at Object.compile (/Applications/Atom.app/Contents/Resources/app/src/package-transpilation-registry.js:79:33)
    at compileFileAtPath (/Applications/Atom.app/Contents/Resources/app/src/compile-cache.js:97:37)
    at Object.value [as .ts] (/Applications/Atom.app/Contents/Resources/app/src/compile-cache.js:238:26)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (/app.asar/static/index.js:47:45)
    at require (internal/module.js:20:19)
    at customRequire (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:96:26)
    at Package.module.exports.Package.requireMainModule (/Applications/Atom.app/Contents/Resources/app/src/package.js:903:35)
    at /Applications/Atom.app/Contents/Resources/app/src/package.js:158:34
    at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app/src/package.js:99:21)
    at Package.module.exports.Package.load (/Applications/Atom.app/Contents/Resources/app/src/package.js:144:18)
    at PackageManager.loadAvailablePackage (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:617:16)
    at config.transact (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:530:20)
    at Config.module.exports.Config.transact (/Applications/Atom.app/Contents/Resources/app/src/config.js:342:22)
    at PackageManager.loadPackages (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:528:23)
    at /Applications/Atom.app/Contents/Resources/app/src/atom-environment.js:821:34
```

**Comment by rsese** [comment]
Thanks @mattdiamond - can you copy the exception from the issue body in https://github.com/jakesankey/line-diff-details/issues/45 to the issue body here?

**Comment by mattdiamond** [comment]
Sure thing.

**Comment by damieng** [comment]
@iolsen this is actually about the typescript transpilation rather than the syntax highlighting or ide work.

There was some discussions recently as to whether we wanted to bump this up or gradually deprecate it in favor of the `"atomTranspilers` system @BinaryMuse put in place.

**Comment by suda** [comment]
For now, the solution is to use `atomTranspilers` and `atom-ts-transpiler` package: https://www.npmjs.com/package/atom-ts-transpiler#setup--configuration

**Comment by smhxx** [comment]
My curiosity led me to dig a little deeper on this, and I've realized that my original explanation of what's causing this error may not be correct; I had thought it was in some way related to the changes made in [this commit on Jul 27, 2015](https://github.com/Microsoft/TypeScript/commit/ad449ce0fae354df03e97e7e8507b9e4f1880001), but I just realized that this change wasn't introduced until TypeScript 1.6, and since `typescript-simple` specifies `typescript ~1.4.1` as a dependency, not `typescript ^1.4.1`, Atom shouldn't be using that version at all. Perhaps this issue was introduced more recently than that?

That would certainly explain why jakesankey/line-diff-details, a package apparently used by 12,000 people, didn't have the bug reported until last month. (jakesankey/line-diff-details#15)
